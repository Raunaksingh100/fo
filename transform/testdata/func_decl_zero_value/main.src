package main

func A[T]() T {
	var x T
	return x
}

func B[T]() []T {
  var x []T
  return x
}

func C[T]() [3]T {
  var x [3]T
  return x
}

func D[T]() map[T]T {
  var x map[T]T
  return x
}

func E[T]() map[T]string {
  var x map[T]string
  return x
}

func F[T]() map[string]T {
  var x map[string]T
  return x
}

func G[T]() *T {
  var x *T
  return x
}

func H[T]() ***T {
  var x ***T
  return x
}

func I[T]() chan T {
  var x chan T
  return x
}

func J[T]() []map[T]chan T {
  var x []map[T]chan T
  return x
}

func K[T]() func(T) {
  var x func(T)
  return x
}

func L[T, U]() map[T]U {
  var x map[T]U
  return x
}

func M[T, U]() func(T) U {
  var x func(T) U
  return x
}

func main() {
}
