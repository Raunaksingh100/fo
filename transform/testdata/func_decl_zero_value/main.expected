package main

import "reflect"

func A(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.TypeOf(T)).Interface()
		return x
	}
}

func B(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.SliceOf(reflect.TypeOf(T))).Interface()
		return x
	}
}

func C(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.ArrayOf(3, reflect.TypeOf(T))).Interface()
		return x
	}
}

func D(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.MapOf(reflect.TypeOf(T), reflect.TypeOf(T))).Interface()
		return x
	}
}

func E(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.MapOf(reflect.TypeOf(T), reflect.TypeOf(""))).Interface()
		return x
	}
}

func F(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.MapOf(reflect.TypeOf(""), reflect.TypeOf(T))).Interface()
		return x
	}
}

func G(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.PtrTo(reflect.TypeOf(T))).Interface()
		return x
	}
}

func H(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.PtrTo(reflect.PtrTo(reflect.PtrTo(reflect.TypeOf(T))))).Interface()
		return x
	}
}

func I(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.ChanOf(reflect.BothDir, reflect.TypeOf(T))).Interface()
		return x
	}
}

func J(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.SliceOf(reflect.MapOf(reflect.TypeOf(T), reflect.ChanOf(reflect.BothDir, reflect.TypeOf(T))))).Interface()
		return x
	}
}

func K(T interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.FuncOf([]reflect.Type{reflect.TypeOf(T)}, []reflect.Type{}, false)).Interface()
		return x
	}
}

func L(T interface{}, U interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.MapOf(reflect.TypeOf(T), reflect.TypeOf(U))).Interface()
		return x
	}
}

func M(T interface{}, U interface{}) func() interface{} {
	return func() interface{} {
		var x = reflect.Zero(reflect.FuncOf([]reflect.Type{reflect.TypeOf(T)}, []reflect.Type{reflect.TypeOf(U)}, false)).Interface()
		return x
	}
}

func main() {
}
